### ISSUE TITLE
Default whitelist: Add 'NixOS'

### ISSUE THREAD

**Issue by axelkar** [rust-lang/rust-clippy#15360]
### Summary

NixOS is a popular Linux distribution, and its name is not a Rust identifier, i.e. it should not be put into `backticks`.

Copy of https://github.com/rust-lang/rust-clippy/issues/10998

This contains an acronym by the specification in https://github.com/rust-lang/rust-clippy/issues/1136

### Lint Name

clippy::doc_markdown

### Reproducer

I tried this code:

```rust
#![warn(clippy::doc_markdown)]
//! example NixOS blah blah
```

I saw this happen:

```
warning: item in documentation is missing backticks
 --> src/lib.rs:2:13
  |
2 | //! example NixOS blah blah
  |             ^^^^^
```

I expected to see this happen:

No warnings

### Version

```text
clippy 0.1.88 (2025-06-23 6b00bc3880)
host: play.rust-lang.org?
release: stable 1.88.0
```

**Comment by axelkar** [comment]
That was super fast! Thank you!

**Comment by samueltardieu** [comment]
You're welcome. And this was written and tested… on NixOS.
